generator/internal/sidekick/command.go

```diff
@@ -0,0 +1,240 @@
+	c.flags.Func(name, usage, fn)
+	return c
+}
+
+// allFlags returns all flags added to this command, as well as its parent hierarchy
+func (c *command) allFlags() []*flag.Flag {
+	var flags []*flag.Flag
+	c.visitAllFlags(func(f *flag.Flag) {
+		flags = append(flags, f)
+	})
+	return flags
+}
+
+// visitAllFlags visits all flags in the command, including those of its parent hierarchy
+func (c *command) visitAllFlags(fn func(f *flag.Flag)) {
+	c.flags.VisitAll(fn)
+	if c.parent != nil {
+		c.parent.visitAllFlags(fn)
+	}
+}
+
+func (c *command) run(rootConfig *Config, cmdLine *CommandLine) error {
+	if c.action == nil {
+		return fmt.Errorf("command %s is not runnable", c.longName())
+	}
+	return c.action(rootConfig, cmdLine)
+}
+
+func (c *command) parseCmdLine(args []string) (*CommandLine, error) {
+	if c.parent != nil {
+		c.parent.visitAllFlags(func(f *flag.Flag) {
+			c.flags.Var(f.Value, f.Name, f.Usage)
+		})
+	}
+
+	if err := c.flags.Parse(args); err != nil {
+		return nil, err
+	}
+
+	return &CommandLine{
+		Command:             args,
+		ProjectRoot:         flagProjectRoot,
+		SpecificationFormat: format,
+		SpecificationSource: source,
+		ServiceConfig:       serviceConfig,
+		Source:              sourceOpts,
+		Language:            flagLanguage,
+		Output:              output,
+		Codec:               codecOpts,
+		DryRun:              dryrun,
+	}, nil
+}
+
+func newCommand(
+	usageLine string,
+	shortDescription string,
+	longDescription string,
+	parent *command,
+	action func(rootConfig *Config, cmdLine *CommandLine) error,
+) *command {
+	if len(usageLine) == 0 {
+		panic("command usage line cannot be empty")
+	}
+	if !strings.HasPrefix(usageLine, "sidekick") {
+		panic(fmt.Sprintf("command usage line must start with sidekick, got: %s", usageLine))
+	}
+	if len(shortDescription) == 0 {
+		panic("command short description cannot be empty")
+	}
+
+	c := &command{
+		usageLine:        usageLine,
+		altNames:         []string{},
+		shortDescription: shortDescription,
+		longDescription:  longDescription,
+		action:           action,
+		flags:            flag.NewFlagSet(usageLine, flag.ContinueOnError),
+		commands:         []*command{},
+		parent:           parent,
+	}
+
+	if parent != nil {
+		parent.commands = append(parent.commands, c)
+	} else if c.name() != "sidekick" {
+		panic("Only the sidekick root command can have a nil parent")
+	}
+
+	c.flags.Usage = func() {
+		c.printUsage()
+	}
+
+	return c
+}
+
+// printUsage prints the usage of the command to os.Stdout, following the same logic as the usageTemplate, but using standard fmt.Println statements instead.
+func (c *command) printUsage() {
+
+	// first prints the entire Long Description for the given command
+	if len(c.longDescription) > 0 {
+		fmt.Println(c.longDescription)
+	} else {
+		fmt.Println(c.shortDescription)
+	}
+	fmt.Printf("\n")
+
+	// Then prints the usage line, skipping over <command> if there are no sub-commands, and [flags] if there are no flags
```

Comment:
nit: some of the comments here start with uppercase, and some are not.  I think generally we have been using full sentences, with a period and everything.

Reply:
I removed some of these as they're not that useful anymore. I had carried them over from the previous template-based implementation. The new impl is easy enough to follow without comments.